src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1673,5 +1682,27 @@ internal override bool HasAsyncMethodBuilderAttribute(out TypeSymbol builderArgu
             builderArgument = _containingType.ContainingPEModule.TryDecodeAttributeWithTypeArgument(this.Handle, AttributeDescription.AsyncMethodBuilderAttribute);
             return builderArgument is not null;
         }
+
+        internal override int? TryGetOverloadResolutionPriority()
+        {
+            if (!_packedFlags.IsOverloadResolutionPriorityPopulated)
+            {
+                if (_containingType.ContainingPEModule.Module.TryGetOverloadResolutionPriorityValue(_handle, out int priority))
+                {
+                    Interlocked.CompareExchange(ref AccessUncommonFields()._lazyOverloadResolutionPriority, priority, 0);
+                }
+#if DEBUG
+                else
+                {
+                    // 0 is the default if nothing is present in metadata, and we don't care about preserving the difference between "not present" and "set to the default value".
```

Comment:
>we don't care about preserving the difference between "not present" and "set to the default value" [](http://example.com/codeflow?start=79&length=98)

In that case, should the virtual method be `int GetOverloadResolutionPriority()` instead, so callers don't rely on the distinction between `null` and `0`?